package/MDAnalysis/analysis/align.py

```diff
@@ -1354,20 +1354,24 @@ def get_atoms_byres(g, match_mask=np.logical_not(mismatch_mask)):
                           "Try to improve your selections for mobile and reference.")
                 logger.error(errmsg)
                 raise SelectionError(errmsg)
-                
+
     if match_atoms:
         # check again because the residue matching heuristic is not very
         # good and can easily be misled (e.g., when one of the selections
         # had fewer atoms but the residues in mobile and reference have
         # each the same number)
-        try:
-            mass_mismatches = (np.absolute(ag1.masses - ag2.masses) > tol_mass)
-        except ValueError:
-            errmsg = ("Failed to find matching atoms: len(reference) = {}, len(mobile) = {} " +
-                    "Try to improve your selections for mobile and reference.").format(
-                        ag1.n_atoms, ag2.n_atoms)
-            logger.error(errmsg)
-            raise_from(SelectionError(errmsg), None)
+        if not hasattr(ag1, 'masses') and hasattr(ag2, 'masses'):
```

Comment:
I think you missed a `not` missing for the second AtomGroup. Currently this will not work.

Reply:
Thank you to point it out :)